task/upload-pypi/0.1/samples/run.yaml

```diff
@@ -54,6 +54,10 @@ kind: PipelineRun
 metadata:
   name: publish-package-pipeline-run
 spec:
+  podTemplate:
+    securityContext:
+      runAsNonRoot: false
+      runAsUser: 0
```

Comment:
Why would you need to run this as root?

Is that because of the python eggs cache trying to write to / ?

What abotu something like this ? https://stackoverflow.com/a/2193746

Reply:
No, I don't think that's the issue. I tried those and they don't seem to work. In this case, here's the error I see:

```
[upload-pypi-run : build-package] running egg_info
[upload-pypi-run : build-package] error: could not create 'dlorenc_requests.egg-info': Permission denied
```

The directory created by the previous git-clone task is created by root, and is not world-writeable. Adding an `ls -l` before running shows that the directory is root owned:

```
[upload-pypi-run : build-package] + ls -la .
[upload-pypi-run : build-package] total 192
[upload-pypi-run : build-package] drwxr-xr-x 9 root root  4096 Jul 25 14:07 .
```

I actually don't know how this is supposed to work with the new git-clone task - maybe we should add a line there to set the permissions world-writeable?

Reply:
Trying to run the entire pipeline as non-root doesn't work either:

```
[fetch-repository : clone] + CHECKOUT_DIR=/workspace/output/
[fetch-repository : clone] + '[[' true '==' true ]]
[fetch-repository : clone] + cleandir
[fetch-repository : clone] + '[[' -d /workspace/output/ ]]
[fetch-repository : clone] + rm -rf /workspace/output//lost+found
```

The clone step fails because it tries to clean up the directory first, and that is root owned.